src/main/java/seedu/toluist/model/TodoList.java

```diff
@@ -57,4 +57,28 @@ public void update(int index, String description) {
             allTasks.get(index).description = description;
         }
     }
+    
+    public ArrayList<Task> getTasksWhosDescriptionContains(String comparison) {
+        ArrayList<Task> foundList = new ArrayList<Task>();
+        
+        for (int i = 0; i < allTasks.size(); i++) {
+            if (allTasks.get(i).isStringContainedInDescriptionIgnoreCase(comparison)) {
+                foundList.add(allTasks.get(i));
+            }
+        }
+        
+        return foundList;
+    }
+
+    public ArrayList<Task> getTasksWhosTagsContains(String comparison) {
+        ArrayList<Task> foundList = new ArrayList<Task>();
+        
+        for (int i = 0; i < allTasks.size(); i++) {
+            if (allTasks.get(i).isStringContainedInAnyTagIgnoreCase(comparison)) {
+                foundList.add(allTasks.get(i));
+            }
+        }
```

Comment:
Notice that the structures of this method and the method above it are exactly the same. So you can do refactoring further by passing in just a generic predicate